From d5945cfa05deba62d79d9f75ab3308047f459c51 Mon Sep 17 00:00:00 2001
From: Tomas Hlavacek <tmshlvck@gmail.com>
Date: Thu, 16 Jun 2016 17:51:26 +0200
Subject: [PATCH] Turris Omnia: Add Omnia LED driver to DT

Add configuration for the leds-omnia driver.

Use the labels from enclosure as the LED names. Use I2C address 0x2b.

The MCU on Omnia board have three I2C slave addresses: 0x2a, 0x2b and 0x2c.

0x2a should be used from userspace applications
0x2b should be used from kernel LED driver
0x2c is the special address for bootloader image transfer
---
 .../arm/boot/dts/armada-385-turris-omnia.dtsi | 73 +++++++++++++++++++
 1 file changed, 73 insertions(+)

diff --git a/arch/arm/boot/dts/armada-385-turris-omnia.dtsi b/arch/arm/boot/dts/armada-385-turris-omnia.dtsi
index 5045015ce823..d968cda97113 100644
--- a/arch/arm/boot/dts/armada-385-turris-omnia.dtsi
+++ b/arch/arm/boot/dts/armada-385-turris-omnia.dtsi
@@ -143,6 +143,79 @@
 				 * 	};
 				 */
 			};
+
+			leds@2b {
+				compatible = "turris-leds,omnia";
+				reg = <0x2b>;
+
+				#address-cells = <1>;
+				#size-cells = <0>;
+
+				led0 {
+					label = "user2";
+					reg = <0x0>;
+				};
+
+				led1 {
+					label = "user1";
+					reg = <0x1>;
+				};
+
+				led2 {
+					label = "pci3";
+					reg = <0x2>;
+				};
+
+				led3 {
+					label = "pci2";
+					reg = <0x3>;
+				};
+
+				led4 {
+					label = "pci1";
+					reg = <0x4>;
+				};
+
+				led5 {
+					label = "wan";
+					reg = <0x5>;
+				};
+
+				led6 {
+					label = "lan4";
+					reg = <0x6>;
+				};
+
+				led7 {
+					label = "lan3";
+					reg = <0x7>;
+				};
+
+				led8 {
+					label = "lan2";
+					reg = <0x8>;
+				};
+
+				led9 {
+					label = "lan1";
+					reg = <0x9>;
+				};
+
+				led10 {
+					label = "lan0";
+					reg = <0xa>;
+				};
+
+				led11 {
+					label = "power";
+					reg = <0xb>;
+				};
+
+		                led12 {
+					label = "all";
+					reg = <0xc>;
+				};
+			};
 		};
 
 		i2c@1 {
